docs/mpi_functions.rst

```diff
@@ -0,0 +1,144 @@
+MPI Functions
+-------------
+
+|MPIFunction|_ is the solution for executing MPI applications remotely using Globus Compute.
+As an extension to |ShellFunction|_, the |MPIFunction|_ supports the same interface
+to specify the command to invoke on the endpoint as well as the capture of output
+streams. However, |MPIFunction|_ diverges from |ShellFunction|_ in the following ways:
+
+1. An |MPIFunction|_ requires a specification of the resources for its execution. This
+   specification must be set on the executor on the client-side. The ``resource_specification: dict``
+   takes the following options:
+
+   .. code-block:: python
+
+        executor.resource_specification = {
+            'num_nodes': <int>,        # Number of nodes required for the application instance
+            'ranks_per_node': <int>,   # Number of ranks / application elements to be launched per node
+            'num_ranks': <int>,        # Number of ranks in total
+        }
+
+2. |MPIFunction|_ is designed to be used with ``GlobusMPIEngine`` on the endpoint.
+   ``GlobusMPIEngine`` is required for the partitioning of a batch job (blocks) dynamically
+   based on the ``resource_specification`` of the |MPIFunction|_.
+
+
+3. |MPIFunction|_ automatically prefixes the supplied command with ``$PARSL_MPI_PREFIX``
+   which resolves to an appropriate mpi launcher prefix (for e.g, ``mpiexec -n 4 -host <NODE1,NODE2>``).
+
+.. warning::
+   |MPIFunction|_ allows for multi-line commands, however the MPI launcher prefix is applied only to
+   the entire command. If multiple commands need to be launched, explicitly use the ``$PARSL_MPI_PREFIX``.
+
+Here is an example configuration for Polaris@ALCF:
+
+.. code-block:: yaml
+
+    # Example configuration from Polaris
+    display_name: PolarisMPI@ALCF
```

Comment:
In order to not imply favoritism, I suggest that we don't use `Polaris` or `Expanse` here.  I understand that these are intended to complete examples, but I don't think they're necessary either.  That is, the only changes from the "standard" configuration are the `type` and `mpi_launcher`, right?

And that suggests that a way to get up and running for users is to first prove that their configuration is valid via "the normal" path, and only then modify for `GlobusMPIEngine`.

Reply:
I've updated this to read as examples for a Slurm-based machine and a PBSPro-based machine.